druid/examples/game_of_life.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Game of life
+
+use druid::kurbo::BezPath;
+use druid::piet::{FontBuilder, ImageFormat, InterpolationMode, Text, TextLayoutBuilder};
+
+
+use druid::{Affine, AppLauncher, BoxConstraints, Color, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx, LocalizedString, PaintCtx, Point, Rect, RenderContext, Size, UpdateCtx, Widget, WindowDesc, Data};
+
+
+const POOL_SIZE: i32 = 1600;
+const CELL_COLOR: Color = Color::rgb8(0xf3 as u8, 0xf4 as u8, 8 as u8);
+
+#[derive(Clone)]
+struct AppData {
+    grid: [i32; 1600],
+}
+
+impl AppData {
+    pub fn evolve(&mut self) {
+        let mut indices_to_mutate: Vec<usize> = vec![];
+        for i in 0..POOL_SIZE {
+            let neighbors = AppData::neighbors(i);
+            let mut sum = 0;
+            for nb_i in &neighbors {
```

Comment:
I would also be tempted to make this an iterator that can filter out out-of-bounds neighbours; or I might at least write,

```rust
for nb_i in neighbors.iter().filter(|n| n < POOL_SIZE) { ... }
```